File: Aliucord/src/main/java/com/aliucord/wrappers/GuildMemberWrapper.java
/*
 * This file is part of Aliucord, an Android Discord client mod.
 * Copyright (c) 2021 Juby210 & Vendicated
 * Licensed under the Open Software License version 3.0
 */

package com.aliucord.wrappers;

import androidx.annotation.Nullable;

import com.discord.api.guildmember.GuildMember;
import com.discord.api.user.User;
import com.discord.api.utcdatetime.UtcDateTime;

import java.util.List;

/**
 * Wraps the obfuscated {@link GuildMember} class to provide nice method names and require only one central
 * update if method names change after an update
 */
@SuppressWarnings("unused")
public class GuildMemberWrapper {
    private final GuildMember guildMember;

    public GuildMemberWrapper(GuildMember guildMember) {
        this.guildMember = guildMember;
    }

    /** Returns the raw (obfuscated) {@link GuildMember} Object associated with this wrapper */
    public final GuildMember raw() {
        return guildMember;
    }

    @Nullable
    public final String getAvatar() {
        return StaticGuildMemberWrapper.getAvatar(guildMember);
    }

    public final long getGuildId() {
        return StaticGuildMemberWrapper.getGuildId(guildMember);
    }

    public final UtcDateTime getJoinedAt() {
        return StaticGuildMemberWrapper.getJoinedAt(guildMember);
    }

    @Nullable
    public final String getNick() {
        return StaticGuildMemberWrapper.getNick(guildMember);
    }

    public final boolean getIsPending() {
        return StaticGuildMemberWrapper.isPending(guildMember);
    }

    @Nullable
    public final String getPremiumSince() {
        return StaticGuildMemberWrapper.getPremiumSince(guildMember);
    }

    public final List<Long> getRoles() {
        return StaticGuildMemberWrapper.getRoles(guildMember);
    }

    public final User getUser() {
        return StaticGuildMemberWrapper.getUser(guildMember);
    }

    public final Long getUserId() {
        return StaticGuildMemberWrapper.getUserId(guildMember);
    }



    /** @deprecated Use {@link StaticGuildMemberWrapper#getAvatar(GuildMember)} instead */
    @Deprecated
    @Nullable
    public static String getAvatar(GuildMember guildMember) {
        return StaticGuildMemberWrapper.getAvatar(guildMember);
    }

    /** @deprecated Use {@link StaticGuildMemberWrapper#getGuildId(GuildMember)} instead */
    @Deprecated
    public static long getGuildId(GuildMember guildMember) {
        return StaticGuildMemberWrapper.getGuildId(guildMember);
    }

    /** @deprecated Use {@link StaticGuildMemberWrapper#getJoinedAt(GuildMember)} instead */
    @Deprecated
    public static UtcDateTime getJoinedAt(GuildMember guildMember) {
        return StaticGuildMemberWrapper.getJoinedAt(guildMember);
    }

    /** @deprecated Use {@link StaticGuildMemberWrapper#getNick(GuildMember)} instead */
    @Deprecated
    @Nullable
    public static String getNick(GuildMember guildMember) {
        return StaticGuildMemberWrapper.getNick(guildMember);
    }

    /** @deprecated Use {@link StaticGuildMemberWrapper#isPending(GuildMember)} instead */
    @Deprecated
    public static boolean getIsPending(GuildMember guildMember) {
        return StaticGuildMemberWrapper.isPending(guildMember);
    }

    /** @deprecated Use {@link StaticGuildMemberWrapper#getPremiumSince(GuildMember)} instead */
    @Deprecated
    @Nullable
    public static String getPremiumSince(GuildMember guildMember) {
        return StaticGuildMemberWrapper.getPremiumSince(guildMember);
    }

    /** @deprecated Use {@link StaticGuildMemberWrapper#getRoles(GuildMember)} instead */
    @Deprecated
    public static List<Long> getRoles(GuildMember guildMember) {
        return StaticGuildMemberWrapper.getRoles(guildMember);
    }

    /** @deprecated Use {@link StaticGuildMemberWrapper#getUser(GuildMember)} instead */
    @Deprecated
    public static User getUser(GuildMember guildMember) {
        return StaticGuildMemberWrapper.getUser(guildMember);
    }

    /** @deprecated Use {@link StaticGuildMemberWrapper#getUserId(GuildMember)} instead */
    @Deprecated
    public final Long getUserId(GuildMember guildMember) {
        return StaticGuildMemberWrapper.getUserId(guildMember);
    }
}

File: Aliucord/src/main/java/com/aliucord/wrappers/GuildMemberWrapper.kt
/*
 * This file is part of Aliucord, an Android Discord client mod.
 * Copyright (c) 2021 Juby210 & Vendicated
 * Licensed under the Open Software License version 3.0
 */

package com.aliucord.wrappers

import com.discord.api.guildmember.GuildMember
import com.discord.api.presence.Presence
import com.discord.api.user.User
import com.discord.api.utcdatetime.UtcDateTime

/**
 * Wraps the obfuscated [GuildMember] class to provide nice method names and require only one central
 * update if method names change after an update
 */
@Suppress("unused")
class GuildMemberWrapper(private val guildMember: GuildMember) {
  /** Returns the raw (obfuscated) [GuildMember] Object associated with this wrapper */
  fun raw() = guildMember

  val avatar
    get() = guildMember.avatar

  val banner
    get() = guildMember.banner

  val bio
    get() = guildMember.bio

  val guildId
    get() = guildMember.guildId

  val joinedAt
    get() = guildMember.joinedAt

  val nick
    get() = guildMember.nick

  val isPending
    get() = guildMember.isPending

  val premiumSince
    get() = guildMember.premiumSince

  val presence
    get() = guildMember.presence

  val roles
    get() = guildMember.roles

  val user
    get() = guildMember.user

  val userId
    get() = guildMember.userId

  companion object {
    @JvmStatic
    val GuildMember.avatar: String?
      get() = b()

    @JvmStatic
    val GuildMember.banner: String?
      get() = c()

    @JvmStatic
    val GuildMember.bio: String?
      get() = d()

    @JvmStatic
    val GuildMember.guildId
      get() = e()

    @JvmStatic
    val GuildMember.joinedAt: UtcDateTime?
      get() = f()

    @JvmStatic
    val GuildMember.nick: String?
      get() = g()

    @JvmStatic
    val GuildMember.isPending
      get() = h()

    @JvmStatic
    val GuildMember.premiumSince: String?
      get() = i()

    @JvmStatic
    val GuildMember.presence: Presence?
      get() = j()

    @JvmStatic
    val GuildMember.roles: List<Long>
      get() = k()

    @JvmStatic
    val GuildMember.user: User?
      get() = l()

    @JvmStatic
    val GuildMember.userId: Long?
      get() = m()
  }
}
